python3Packages.sapi-python-client: init at 0.13.1

MrMebelMan opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

MrMebelMan committed:
python3Packages.sapi-python-client: init at 0.13.1

jonringer commented on pkgs/development/python-modules/sapi-python-client/default.nix:
```diff
@@ -0,0 +1,28 @@
+
+    nativeBuildInputs = [ git ]; 
+
+    propagatedBuildInputs = [ setuptools setuptools_scm ]; 
```
setuptools_scm should be in native inputs

jonringer commented on pkgs/development/python-modules/sapi-python-client/default.nix:
```diff
@@ -0,0 +1,28 @@
+        url = "https://github.com/keboola/sapi-python-client.git"; 
+        sha256 = "1zfxz9066ddpdmyxrwvns2v8kjn5dyyqgbiaai05k4ksrrf3zv22"; 
+        rev = "0.1.3"; 
+    }; 
```
please use github helper, pulls down just a tarball instead of git history
```suggestion
    src = fetchFromGitHub { 
        owner = "keboola";
        repo = pname;
        rev  = version;
        sha256 = "1zfxz9066ddpdmyxrwvns2v8kjn5dyyqgbiaai05k4ksrrf3zv2a"; 
    }; 
```
sha isn't right, but tha tcan be fixed

jonringer commented on pkgs/development/python-modules/sapi-python-client/default.nix:
```diff
@@ -0,0 +1,28 @@
+{ stdenv, git, setuptools, setuptools_scm, fetchgit, requests, boto3, buildPythonPackage, responses }: 
```
```suggestion
{ stdenv, git, setuptools, setuptools_scm, fetchFromGitHub, requests, boto3, buildPythonPackage, responses }: 
```

jonringer commented on pkgs/development/python-modules/sapi-python-client/default.nix:
```diff
@@ -0,0 +1,28 @@
+
+    propagatedBuildInputs = [ setuptools setuptools_scm ]; 
+
+    buildInputs = [ requests boto3 responses ];
```
python packages should go in propagatedBuildInputs
```suggestion
    buildInputs = [ requests boto3 responses ];
```

MrMebelMan committed:
Move setuptools_scm to nativeBuildInputs

use fetchFromGitHub instead of fetchgit

move Python packages to propagatedBuildInputs

MrMebelMan commented:
Updated as per review comments, but now there's error I can't figure out...

```
LookupError: setuptools-scm was unable to detect version for '/build/source'.

Make sure you're either building from a fully intact git repository or PyPI tarballs. Most other sources (such as GitHub's tarballs, a git checkout without the .git folder) don't contain the necessary metadata and will not work.

For example, if you're using pip, instead of https://github.com/user/proj/archive/master.zip use git+https://github.com/user/proj.git#egg=proj
```

mmahut commented:
Maybe try something like: https://github.com/NixOS/nixpkgs/blob/eed30dfa5c87e6aca95850b1e16da4113275084a/pkgs/development/python-modules/measurement/default.nix#L18

MrMebelMan commented:
> Maybe try something like:
> 
> https://github.com/NixOS/nixpkgs/blob/eed30dfa5c87e6aca95850b1e16da4113275084a/pkgs/development/python-modules/measurement/default.nix#L18

Oh 👀
Thanks!

MrMebelMan committed:
Use a specific version in setup.py, instead of the scm_version

MrMebelMan commented:
Rebased master.

mmahut commented:
@GrahamcOfBorg build python3Packages.sapi-python-client
